id: retry_demo
namespace: dev

tasks:
  - id: fail_four_times
    type: io.kestra.plugin.scripts.shell.Commands
    runner: PROCESS
    commands:
      - 'if [ "{{taskrun.attemptsCount }}" -eq 4]; then exit 0; else exit 1; fi'
    retry:
      type: constant
      interval: PT2S
      maxAttempt: 5
      maxDuration: PT1M
      warningOnRetry: false

errors:
  - id: will_never_run
    type: io.kestra.plugin.core.debug.Return
    format: This will never be executed as retries will fix the issue
